src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,13 +45,48 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        setAddress();
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Sets the address objects for block, street, unit, postal
+     */
+    private void setAddress() {
+        String[] splitAddress = value.split(ADDRESS_DELIMITER);
+        setBlock(new AddressBlock(splitAddress[ADDRESS_BLOCK_INDEX].trim()));
+        setStreet(new AddressStreet(splitAddress[ADDRESS_STREET_INDEX].trim()));
+        setUnit(new AddressUnit(splitAddress[ADDRESS_UNIT_INDEX].trim()));
+        setPostalCode(new AddressPostalCode(splitAddress[ADDRESS_POSTAL_CODE_INDEX].trim()));
+    }
+
+    /**
+     * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        String testSplit[] = test.split(ADDRESS_DELIMITER);
+        return hasCorrectParameters(testSplit) && isValidPostalCodeLength(testSplit) ;
+    }
+
+    /**
+     * Checks if the postal code has valid postal code length
+     * @param testSplit splitted string array for the address
+     * @return true if postal code length is 6
+     */
+    private static boolean isValidPostalCodeLength(String[] testSplit) {
+        if (testSplit.length <= 3) { //Insufficient parameters, will not contain postal in the correct section.
```

Comment:
Is there a real reason for checking for correct parameter numbers here?
I believe you've checked it in `hasCorrectParameters(...)`.
If `hasCorrectParameters(...)` returns false due to insufficient parameters, `isValidPostalCodeLength(...)` will not be entered due to the property of `&&` right?